requestor-tutorials/service-development/service-example-2-managed-erigon.md

```diff
@@ -38,7 +38,272 @@ A typical interaction flow proceeds in the following manner:
 
 ## Runtime implementation
 
+Here we demonstrate how self-contained Exe-Unit runtimes can be implemented, tested and finally plugged into the Provider's daemon.
+
+{% hint style="warning" %}
+Please be aware that a custom runtime doesn't provide the level of isolation comparable with a virtual machine. It's just an executable running with the same privileges as yagna daemon.
+{% endhint %}
+
+### Erigon runtime
+
+First, let's discuss what is the purpose of the Erigon runtime to gain some high level overview. The runtime:
+
+* Accepts an Ethereum network's name (e.g. 'mainnet' or 'rinkeby') from the requestor agent,
+* Starts Erigon binaries to sync with the chosen network and serve Erigon RPC endpoint,
+* Generates credentials for basic authentication which is implemented with nginx,
+* Returns the credentials to the requestor agent so the user of the service can connect to the Erigon RPC.
+
+Please refer to the "Provider's node" in the above Architecture overview diagram.
+
+### Implementation
+
+{% hint style="info" %}
+Before diving into this section you might want to review simpler [example runtime](https://github.com/golemfactory/ya-runtime-sdk/blob/main/examples/example-runtime/src/main.rs) available in the [ya-runtime-sdk](https://github.com/golemfactory/ya-runtime-sdk/) repository.
+{% endhint %}
+
+Erigon runtime declares two structures
+
+* `ErigonConf` - containing runtime configuration with default values,
+* `ErigonRuntime` - derives from `RuntimeDef` implements the required methods from `ya-runtime-sdk::Runtime` trait
+
+At least these methods from the `Runtime` trait have to be implemented but SDK provides default implementation to several others (TODO: link runtime SDK docs)
+
+* `deploy` - It's run only once per activity, does not accept parameters (but there are plans to add them in the future release), return value is not passed to the requestor agent.
```

Comment:
For clarity, I strongly recommend making `deploy`, `start`, `run_command` and `stop` into sub-sections of 'Implementation'. These single-bullet lists don't stand out enough.